docs/root/configuration/http/http_filters/adaptive_concurrency_filter.rst

```diff
@@ -0,0 +1,155 @@
+
+Concurrency Controllers
+-----------------------
+Concurrency controllers implement the algorithm responsible for making forwarding decisions for each
+request and recording latency samples to use in the calculation of the concurrency limit.
+
+Gradient Controller
+~~~~~~~~~~~~~~~~~~~
+The gradient controller makes forwarding decisions based on a periodically measured ideal round-trip
+time (minRTT) for an upstream.
+
+:ref:`v2 API reference <envoy_api_msg_config.filter.http.adaptive_concurrency.v2alpha.GradientControllerConfig>`
+
+Calculating the minRTT
+^^^^^^^^^^^^^^^^^^^^^^
+
+The minRTT is periodically measured by only allowing a single
+outstanding request at a time and measuring the latency under these ideal conditions. The length of
+this minRTT calculation window is variable depending on the number of requests the filter is
+configured to aggregate to represent the expected latency of an upstream.
+
+.. note::
+
+    It is possible that there is a noticeable increase in request 503s during the minRTT
+    measurement window because of a significant drop in the concurrency limit. This is expected and
+    it is recommended to use :ref:`the previous_hosts retry predicate <arch_overview_http_retry_plugins>` to avoid unexpected synchonization of upstream hosts configured with the adaptive concurrency filter.
+
+The calculated minRTT is then used in the calculation of a value referred to as the *gradient*.
+
+The Gradient
+^^^^^^^^^^^^
+The gradient is calculated using summarized sampled request latencies (sampleRTT):
+
+.. math::
+
+    gradient = \frac{minRTT}{sampleRTT}
+
+This gradient value has a useful property, such that it decreases as the sampled latencies increase.
+The gradient value is then used to update the concurrency limit via:
+
+.. math::
+
+    limit_{new} = gradient * limit_{old} + headroom
+
+Concurrency Limit Headroom
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+The headroom value is necessary as a driving factor to increase the concurrency limit when the
+sampleRTT is in the same ballpark as the minRTT. This value must be present in the limit
+calculation, since it forces the concurrency limit to increase until there is a deviation from the
+minRTT latency. In the absence of a headroom value, the concurrency limit could potentially stagnate
+at an unnecessary small value if the sampleRTT and minRTT are close to each other.
+
+Because the headroom value is so necessary to the proper function fo the gradient controller, the
+headroom value is unconfigurable and pinned to the square-root of the concurrency limit.
+
+Example Configuration
+---------------------
+An example filter configuration:
+
+.. code-block:: yaml
+
+  name: envoy.filters.http.adaptive_concurrency
+  config:
+    gradient_controller_config:
+      sample_aggregate_percentile:
+        value: 50
+      concurrency_limit_params:
+        concurrency_update_interval: 0.1s
+      min_rtt_calc_params:
+        jitter:
+          value: 15
```

Comment:
i dont see any explanation in the above doc for what this parameter means, might be good to include.